src/Stratis.Bitcoin/Configuration/NodeSettings.cs

```diff
@@ -391,7 +393,43 @@ public static void PrintHelp(Network network)
 
             defaults.Logger.LogInformation(builder.ToString());
 
-            ConnectionManagerSettings.PrintHelp(network);
+            // Print command-line help
+            var objects = new object[] { network, serviceProvider };
+            CallStaticMethod(typeof(ConnectionManagerSettings), "PrintHelp", objects);
```

Comment:
Why are we trying to check if a method exists and if it does, invoke it using reflection? Isn't this what a type system is for? Ie. use an interface.

Reply:
Ah, I see that it's because the method is expected to be static. Feel like that's largely unnecessary and this problem could be solved with an `object is IPrintHelp` or similar.

Reply:
That would probably require changing a bunch of other stuff... so just ignore these comments 😄

Reply:
That's being left for the next PR. :)

Reply:
Yes these are static methods so I am going with this approach for now. Keeping the PR manageable.